internal/prototest/yaml.go

```diff
@@ -0,0 +1,370 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prototest
+
+import (
+	"cmp"
+	"fmt"
+	"math"
+	"slices"
+	"strings"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+const license = `# Copyright 2020-2024 Buf Technologies, Inc.
```

Comment:
This is actually problematic, because if the license checker does start inserting these comments, they will cause tests to fail, because the tests expect byte-for-byte equality.

What I'm going to do is remove the license for now, and when we do add automatic splatting-in of licenses, I'll set things up to delete the comment prefix in the comparator.